frontend/src/metabase-lib/v1/parameters/utils/mbql.js

```diff
@@ -220,33 +221,65 @@ function filterParameterToMBQL(query, stageIndex, parameter) {
     [parameter.target[1]],
   );
   if (columnIndex < 0) {
-    return null;
+    return;
   }
 
-  const column = columns[columnIndex];
-  const fieldRef = Lib.legacyRef(query, stageIndex, column);
+  return columns[columnIndex];
+}
 
+/** compiles a parameter with value to legacy MBQL */
+function filterParameterToMBQL(query, stageIndex, parameter, column) {
```

Comment:
Long-term vision is move everything related to parameter application out of `metabase-lib` and remove any dependency on raw mbql. But for now I'm happy to just make this change for date parameters.